terraform/environments/electronic-monitoring-data/analytical_platform_share.tf

```diff
@@ -0,0 +1,301 @@
+  secret_id = data.aws_secretsmanager_secret.dbt_secrets.id
+
+  depends_on = [aws_secretsmanager_secret.dbt_secrets]
+}
+
+
+# Retrieves the source role of terraform's current caller identity
+data "aws_iam_session_context" "current" {
+  arn = data.aws_caller_identity.current.arn
+}
+
+data "aws_iam_roles" "data_engineering_roles" {
+  name_regex = "AWSReservedSSO_modernisation-platform-4*"
+}
+
+## DBT Analytics EKS Cluster Identifier
+# PlaceHolder Secrets
+resource "aws_secretsmanager_secret_version" "dbt_secrets" {
+  count = local.enable_dbt_k8s_secrets ? 1 : 0
+
+  secret_id     = aws_secretsmanager_secret.dbt_secrets[0].id
+  secret_string = jsonencode(local.dbt_k8s_secrets_placeholder)
+
+  lifecycle {
+    ignore_changes = [secret_string, ]
+  }
+
+  depends_on = [aws_secretsmanager_secret.dbt_secrets]
+}
+
+resource "aws_secretsmanager_secret" "dbt_secrets" {
+  #checkov:skip=CKV2_AWS_57: “Ignore - Ensure Secrets Manager secrets should have automatic rotation enabled"
+  #checkov:skip=CKV_AWS_149: "Ensure that Secrets Manager secret is encrypted using KMS CMK"
+
+  count = local.enable_dbt_k8s_secrets ? 1 : 0
+
+  name = "external/analytics_platform/k8s_dbt_auth"
+
+  recovery_window_in_days = 0
+
+  tags = merge(
+    local.tags,
+    {
+      Name          = "external/cloud_platform/k8s_auth"
+      Resource_Type = "Secrets"
+      Source        = "Analytics-Platform"
+      Jira          = "DPR2-751"
+    }
+  )
+}
+
+# TLS Certificate for OIDC URL, DBT K8s Platform
+data "tls_certificate" "dbt_analytics" {
+  url = "https://oidc.eks.eu-west-2.amazonaws.com/id/${jsondecode(data.aws_secretsmanager_secret_version.dbt_secrets.secret_string)["oidc_cluster_identifier"]}"
+}
+
+
+## OIDC, OpenID Connect
+resource "aws_iam_openid_connect_provider" "cluster" {
+  client_id_list  = ["sts.amazonaws.com"]
+  thumbprint_list = [data.tls_certificate.dbt_analytics.certificates[0].sha1_fingerprint]
+  url             = "https://oidc.eks.eu-west-2.amazonaws.com/id/${jsondecode(data.aws_secretsmanager_secret_version.dbt_secrets.secret_string)["oidc_cluster_identifier"]}"
+}
+
+## Role
+## CrossAccount DataAPI Cross Account Role,
+# CrossAccount DataAPI Assume Policy
+data "aws_iam_policy_document" "dataapi_cross_assume" {
+  #checkov:skip=CKV_AWS_110:Ensure IAM policies does not allow privilege escalation
+  #checkov:skip=CKV_AWS_358:OIDC trust policies only allows actions from a specific known organization Already
+  #checkov:skip=CKV_AWS_107:Ensure IAM policies does not allow credentials exposure
+  #checkov:skip=CKV_AWS_111:Ensure IAM policies does not allow write access without constraints
+  #checkov:skip=CKV_AWS_356
+  #checkov:skip=CKV_AWS_109
+  #checkov:skip=CKV_AWS_1
+  #checkov:skip=CKV_AWS_283
+  #checkov:skip=CKV_AWS_49
+  #checkov:skip=CKV_AWS_108
+
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::754256621582:root"]
```

Comment:
what account it this? Should this be EM?

Reply:
im not sure - think its a cloud platform account